File: LeetCode/Java Solutions/Number of Dice Rolls With Target Sum/number.java
public class Solution {
    public int numRollsToTarget(int n, int k, int target) {
        int MOD = 1000000007;

        // Initialize the dp array.
        int[][] dp = new int[n + 1][target + 1];
        dp[0][0] = 1;  // Base case: 1 way to achieve sum 0 with 0 dice

        for (int dice = 1; dice <= n; dice++) {
            for (int t = 1; t <= target; t++) {
                for (int face = 1; face <= k && face <= t; face++) {
                    dp[dice][t] = (dp[dice][t] + dp[dice - 1][t - face]) % MOD;
                }
            }
        }

        return dp[n][target];
    }
}

File: LeetCode/Python Solutions/Number of Dice Rolls With Target Sum/number.py
class Solution:
    def numRollsToTarget(self, n: int, k: int, target: int) -> int:
        MOD = 10**9 + 7
        
        # Initialize the dp array.
        dp = [[0] * (target + 1) for _ in range(n + 1)]
        dp[0][0] = 1  # Base case: 0 ways to achieve sum 0 with 0 dice
        
        for dice in range(1, n + 1):
            for t in range(1, target + 1):
                for face in range(1, min(k, t) + 1):
                    dp[dice][t] = (dp[dice][t] + dp[dice - 1][t - face]) % MOD
        
        return dp[n][target]
